test/extended/util/framework.go

```diff
@@ -1018,32 +1018,60 @@ func CheckBuildCancelled(b *buildv1.Build) bool {
 	return b.Status.Phase == buildv1.BuildPhaseCancelled
 }
 
-// WaitForServiceAccount waits until the named service account gets fully
-// provisioned
+func getServiceAccount(c corev1client.ServiceAccountInterface, name string) (*corev1.ServiceAccount, error) {
+	sc, err := c.Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil {
+		if kapierrs.IsNotFound(err) || kapierrs.IsForbidden(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	return sc, nil
+}
+
+func serviceAccountContainsPullSecret(sa *corev1.ServiceAccount) bool {
+	hasDockerCfg := false
+	for _, s := range sa.ImagePullSecrets {
+		if strings.Contains(s.Name, "-dockercfg-") {
+			hasDockerCfg = true
+		}
+	}
+	return hasDockerCfg
+}
+
+// WaitForServiceAccountWithSecret waits until the named service account gets fully
+// provisioned. without dockercfg secrets
 func WaitForServiceAccount(c corev1client.ServiceAccountInterface, name string) error {
 	waitFn := func() (bool, error) {
-		sc, err := c.Get(context.Background(), name, metav1.GetOptions{})
+		sa, err := getServiceAccount(c, name)
```

Comment:
I see no reason to change this logic

```go
sc, err := c.Get(context.Background(), name, metav1.GetOptions{})
if err != nil {
	// If we can't access the service accounts, let's wait till the controller
	// create it.
	if kapierrs.IsNotFound(err) || kapierrs.IsForbidden(err) {
		e2e.Logf("Waiting for service account %q to be available: %v (will retry) ...", name, err)
		return false, nil
	}
	return false, fmt.Errorf("Failed to get service account %q: %v", name, err)
}
```

is kube-standard (versus nil, nil return) and does what you're trying to do, right?

Reply:
Was just trying to refactor it to avoid repetition, since the same code was in two different functions. Do you think its best to have it self-contained in each of the functions? It is `WaitForServiceAccount` and `WaitForServiceAccountWithSecret`.

Reply:
still a problem here.